tee-worker/litentry/core/assertion-build/src/nodereal/spaceid/mod.rs

```diff
@@ -0,0 +1,226 @@
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+pub mod bnb_domain_holding_amount;
+
+use crate::*;
+use lc_data_providers::{
+	nodereal::{DomainInfo, NoderealClient, SpaceIDApiList},
+	DataProviderConfigReader, ReadDataProviderConfig,
+};
+use serde::{Deserialize, Serialize};
+use std::string::ToString;
+
+pub struct SPACEIDLogic;
+pub trait SPACEIDLogicInterface {
+	fn get_bnb_domain_holding_amount(addresses: &[String]) -> Result<usize>;
+}
+
+impl SPACEIDLogicInterface for SPACEIDLogic {
+	fn get_bnb_domain_holding_amount(owners: &[String]) -> Result<usize> {
+		let config = DataProviderConfigReader::read()
+			.map_err(|e| Error::RequestVCFailed(Assertion::BnbDomainHolding, e))?;
+		let mut client = NoderealClient::new(&config);
+		let response = client.by_owners(owners).map_err(|e| {
+			Error::RequestVCFailed(Assertion::BnbDomainHolding, e.into_error_detail())
+		})?;
+
+		let owned_domains: Domains = Domains::from_value(&response)
+			.map_err(|_e| Error::UnclassifiedError(ErrorDetail::ParseError))?;
+
+		Ok(owned_domains.non_expired_domains()?.len())
+	}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Domain {
+	owner: String,
+	domain_infos: Vec<DomainInfo>,
+}
+
+impl Domain {
+	pub fn new(owner: String) -> Self {
+		Self { owner, domain_infos: vec![] }
+	}
+
+	pub fn add(&mut self, domain_info: DomainInfo) {
+		self.domain_infos.push(domain_info);
+	}
+
+	pub fn size(&self) -> usize {
+		self.domain_infos.len()
+	}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Domains {
+	domains: Vec<Domain>,
+}
+
+impl Domains {
+	pub fn from_value(value: &serde_json::Value) -> Result<Domains> {
+		let mut domains: Vec<Domain> = vec![];
+
+		if let Some(obj_map) = value.as_object() {
+			for (key, val) in obj_map {
+				let mut owned = Domain::new(key.to_string());
+
+				if let Some(values) = val.as_array() {
+					for info in values {
+						let domain_info: DomainInfo =
+							DomainInfo::from_value(info).map_err(Error::UnclassifiedError)?;
+
+						owned.add(domain_info);
+					}
+				}
+
+				domains.push(owned);
+			}
+		}
+
+		Ok(Domains { domains })
+	}
+
+	pub fn size(&self) -> usize {
+		self.domains.len()
+	}
+
+	pub fn non_expired_domains(&self) -> Result<Vec<&DomainInfo>> {
```

Comment:
it reads like the function should return a list of domains. What do you think about renaming this function to 
```suggestion
	pub fn non_expired_domain_infos(&self) -> Result<Vec<&DomainInfo>> {
```